tests/gocase/unit/keyspace_test.go

```diff
@@ -0,0 +1,206 @@
+		require.Equal(t, int64(3), rdb.Del(ctx, "foo1", "foo2", "foo3").Val())
+		require.Equal(t, []interface{}{nil, nil, nil}, rdb.MGet(ctx, "foo1", "foo2", "foo3").Val())
+	})
+
+	t.Run("KEYS with pattern", func(t *testing.T) {
+		for _, key := range []string{"key_x", "key_y", "key_z", "foo_a", "foo_b", "foo_c"} {
+			require.NoError(t, rdb.Set(ctx, key, "hello", 0).Err())
+		}
+		keys := rdb.Keys(ctx, "foo*").Val()
+		sort.Slice(keys, func(i, j int) bool {
+			return keys[i] < keys[j]
+		})
+		require.Equal(t, []string{"foo_a", "foo_b", "foo_c"}, keys)
+	})
+
+	t.Run("KEYS to get all keys", func(t *testing.T) {
+		keys := rdb.Keys(ctx, "*").Val()
+		sort.Slice(keys, func(i, j int) bool {
+			return keys[i] < keys[j]
+		})
+		require.Equal(t, []string{"foo_a", "foo_b", "foo_c", "key_x", "key_y", "key_z"}, keys)
+	})
+
+	t.Run("DEL all keys", func(t *testing.T) {
+		vals := rdb.Keys(ctx, "*").Val()
+		require.Equal(t, int64(len(vals)), rdb.Del(ctx, vals...).Val())
+		require.NoError(t, rdb.Do(ctx, "dbsize", "scan").Err())
+		time.Sleep(100 * time.Millisecond)
+		require.Equal(t, int64(0), rdb.Do(ctx, "dbsize").Val())
+	})
+
+	t.Run("EXISTS", func(t *testing.T) {
+		newKey := "newkey"
+		require.NoError(t, rdb.Set(ctx, newKey, "test", 0).Err())
+		require.Equal(t, int64(1), rdb.Exists(ctx, newKey).Val())
+		require.Equal(t, int64(1), rdb.Del(ctx, newKey).Val())
+		require.Equal(t, int64(0), rdb.Exists(ctx, newKey).Val())
+	})
+
+	t.Run("Zero length value in key. SET/GET/EXISTS", func(t *testing.T) {
+		emptyKey := "emptykey"
+		require.NoError(t, rdb.Set(ctx, emptyKey, nil, 0).Err())
+		require.Equal(t, int64(1), rdb.Exists(ctx, emptyKey).Val())
+		require.Equal(t, int64(1), rdb.Del(ctx, emptyKey).Val())
+		require.Equal(t, int64(0), rdb.Exists(ctx, emptyKey).Val())
+	})
+
+	t.Run("Commands pipelining", func(t *testing.T) {
+		c := srv.NewTCPClient()
+		defer func() { require.NoError(t, c.Close()) }()
+		require.NoError(t, c.Write("SET k1 xyzk\r\nGET k1\r\nPING\r\n"))
+		r, err := c.ReadLine()
+		require.NoError(t, err)
+		require.Equal(t, "+OK", r)
+		r, err = c.ReadLine()
+		require.NoError(t, err)
+		require.Equal(t, "$4", r)
+		r, err = c.ReadLine()
+		require.NoError(t, err)
+		require.Equal(t, "xyzk", r)
+		r, err = c.ReadLine()
+		require.NoError(t, err)
+		require.Equal(t, "+PONG", r)
+	})
+
+	t.Run("Non existing command", func(t *testing.T) {
+		require.Contains(t, rdb.Do(ctx, "foobaredcommand").Err().Error(), "ERR")
+	})
+
+	t.Run("RANDOMKEY", func(t *testing.T) {
+		rdb.FlushDB(ctx)
+		require.NoError(t, rdb.Set(ctx, "foo", "x", 0).Err())
+		require.NoError(t, rdb.Set(ctx, "bar", "y", 0).Err())
+
+		fooSeen := 0
+		barSeen := 0
+		for i := 0; i < 1000; i++ {
+			randomKey := rdb.RandomKey(ctx).Val()
+			if randomKey == "foo" {
+				fooSeen = 1
+			} else if randomKey == "bar" {
+				barSeen = 1
+			}
+			if fooSeen == 1 || barSeen == 1 {
+				break
+			}
+			time.Sleep(10 * time.Millisecond)
```

Comment:
Why do we need to sleep here?

Reply:
We random by time seed, so it may always get the same seed if requests fast enough. But the case looks good until now, so let's remove the sleep.

Reply:
We random by time seed, so it may always get the same seed if requests fast enough. But the case looks good until now, so let's remove the sleep.